test/docs/dash-test-workflow-p4-saithrift.md

```diff
@@ -0,0 +1,89 @@
+* A [P4-DPDK](https://github.com/p4lang/p4-dpdk-target) based software dataplane championed by Intel, using the PNA archtecture. This is much more than a "simulator," it is a first-class, SW dataplane implementation. When executing on a normal CPU (versus a dedicated hardware device) it can serve as a fairly performant DASH dataplane simulator.
+
+### BMv2 Simulator Design Details
+The diagram show a few details about the design of the [bmv2](https://github.com/p4lang/behavioral-model) simulator. The community bmv2 program is modified to support DASH requirements (e.g. stateful connection tracking). The source code assumes an enhanced v1model.
+
+A saithrift server skeleton is linked to a libsai library to yield a saithrift API endoint, used by the test runner.
+
+The bmv2 simulator has a built-in P4Runtime server. A SAI-to-P4Runtime adaptor maps SAI API calls made to the libsai API, into P4Runtime API gRPC calls over a socket to the bmv2 endpoint. SAI object operations are thus converted into P4 entity operations.
+
+> **TODO** Verify that a gRPC interface is used, vs. an internal compiled interface.
+
+The bmv2 simulator is bound to host veth ports at startup.  These are "wired" to the software traffic generator using Linux bridging or similar.
+
+### P4-DPDK Simulator Design Details
+The diagram shows a few details about the design of the [P4-DPDK](https://github.com/p4lang/p4-dpdk-target) dataplane, which as stated previously, is not merely  a simulator, but can be used as one for DASH purposes.
+
+The P4-DPDK dataplane has an internal, compiled [Table-Driven Interface (TDI)](https://github.com/p4lang/tdi). This abstraction is analogous to the Switch Abstraction Interface (SAI), except tailored to P4 Dataplanes.
+
+A libsai library will be developed which maps SAI conceptual APIs into the [sirius_pipeline.p4](../sirius_pipeline.p4) TDI implementation, all done in-process. This is very efficient because it avoids serialization over P4Runtime.
+
+The P4-DPDK dataplane also has a built-in P4Runtime server which can be used develop and test the P4 Model. This is not part of the official DASH test workflow.
+
+The P4-DPDK dataplane is bound to host veth ports at startup.  These are "wired" to the software traffic generator using Linux bridging or similar.
+
+## Summarizing it all
+* Standard DASH test cases and test runners apply conformance and performance tests against software-based DASH dataplane simulators.
+* Software dataplane have lower scale and performance capabilities compared to production hardware implementations. Therefore, test must be marked for suitability on each target to respect performance and scale limitations of each software implementation. The test runners will run only the appropriate subset for a given target.
```

Comment:
s/test must be marked/tests must be marked/  <-- for future reference, this is a SW dev's way to show a string substitution i.e. correct a typo

Reply:
@chrispsommers Could you clarify the remark you made? It is not clear what I need to do. Thanks